File: 231.2-的幂.go
package jp_leetcode

import "math"

/*
 * @lc app=leetcode.cn id=231 lang=golang
 *
 * [231] 2 的幂
 */

// @lc code=start
func isPowerOfTwo(n int) bool {
	var res int = math.MinInt

	for x := 0; res < n; x++ {
		res = int(math.Pow(2.0, float64(x)))
	}

	return res == n

}

// @lc code=end

// func main() {
// 	isPowerOfTwo(1)
// }

File: leetcode_ts/src/231.2-的幂.ts
/*
 * @lc app=leetcode.cn id=231 lang=typescript
 *
 * [231] 2 的幂
 */

// @lc code=start
function isPowerOfTwo(n: number): boolean {
  // 只要是2的幂次方，二进制表示下最高位一定是1，且只有1位是1
  return (/^10*$/.test(n.toString(2)))
};
// @lc code=end

